### PR TITLE
Add sanity test function

### PR DESCRIPTION
**Pull request by jwang078**

### CODE REVIEW

File: tasks/sanity_test.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        pytest.fail("analyze() encountered exception '{}' with test image {}".format(error[0], error[1]))
+
+def run_algorithm(algorithm, test_imgs, num_runs, cconn):
+    """ Wrapper function to run the algorithm on a separate thread. """
+    for i in range(num_runs):
+        try:
+            algorithm.analyze(test_imgs[i], True)
+        except Exception as e:
+            # Send the error message and the image number back to the main thread
+            cconn.send((e, i))
```
Comment (AVSurfer123) [inline tasks/sanity_test.py:63]: You don't seem to do anything with `cconn` back in the main thread. Does it get handled somewhere else?

File: tasks/sanity_test.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        algorithm: object that extends TaskPerceiver
+
+    Example usage:
+        from sanity_test import sanity_test
+        import pytest
+        class Algorithm1(TaskPerceiver):
+            def analyze(self, frame, debug):
+                pass
+        class TestAlgorithm1():
+            @pytest.fixture(scope="class")
```
Comment (AVSurfer123) [inline tasks/sanity_test.py]: I feel like `pytest.mark.parametrize` is better than using fixtures in the example here.

**Review by AVSurfer123 — CHANGES_REQUESTED** [review]

File: tasks/sanity_test.py (hunk 1 above)
Comment (jwang078) [reply tasks/sanity_test.py:63]: cconn and pconn are linked from `pconn, cconn = multiprocessing.Pipe()` so this changes `pconn.poll()` and `pconn.recv()`

File: tasks/sanity_test.py (hunk 2 above)
Comment (jwang078) [reply tasks/sanity_test.py]: I was hoping for people to use the `algorithm` and `test_imgs` fixtures in other test functions too. Do you think it's better to specifically specify `@pytest.mark.parameterize("algorithm,test_imgs", [(Algorithm1(), get_test_imgs())])` for each test?

Comment (AVSurfer123) [reply tasks/sanity_test.py]: That's how I did it during my internship. Then you don't need to deal with creating a test class and fixtures. You just need to provide some inputs and expected results to a test function.

File: tasks/sanity_test.py (hunk 1 above)
Comment (AVSurfer123) [reply tasks/sanity_test.py:63]: Ah I see.

**Review by AVSurfer123 — APPROVED** [review]

**Pull request merged**